stripe/src/test/java/com/stripe/android/paymentsheet/CurrencyFormatterTest.kt

```diff
@@ -1,33 +1,232 @@
 package com.stripe.android.paymentsheet
 
+import android.icu.text.NumberFormat
 import com.google.common.truth.Truth.assertThat
 import org.junit.Test
 import java.util.Currency
 import java.util.Locale
 
+/**
+ * See [currencies](https://stripe.com/docs/currencies#minimum-and-maximum-charge-amounts)
+ * for useful test scenarios.
+ *
+ * Also by looking at the [NumberFormat] fields returned from [NumberFormat#getCurrencyInstance]
+ * we see other things that vary between countries. Each one is not verified here as we are not testing
+ * the library.
+ *
+ * The focus of this testing is to make sure the:
+ *      - amount is formatted with the number of decimal digits based on the amount currency, and
+ *      not the locale currency
+ *      - decimal number is formatted in the locale format
+ *      - currency symbol location is formatted in the locale format
+ *      - currency symbol is formatted in the locale format
+ *
+ * If we added instrumented tests to our build server, this would be a good set of cases to run on device.
+ *
+ * Notable in testing is that if the targetLocale has a language vs a language and
+ * country the outputted results are quite different.
+ *
+ * Similar tests exist on iOS
+ * [here](https://github.com/stripe/stripe-ios/blob/master/Tests/Tests/NSString+StripeTest.swift)
+ */
 class CurrencyFormatterTest {
     private val currencyFormatter = CurrencyFormatter()
 
     @Test
-    fun `format returns expected value`() {
-        Locale.setDefault(Locale.US)
+    fun `amount currency USD, locale US`() {
         assertThat(
             currencyFormatter.format(
-                12355L,
+                123412L,
                 Currency.getInstance("USD")
             )
-        ).isEqualTo("$123.55")
+        ).isEqualTo("$1,234.12")
+    }
 
+    @Test
+    fun `amount currency Germany, locale US`() {
         val euro = Currency.getInstance(Locale.GERMANY)
-        assertThat(currencyFormatter.format(12300L, euro))
-            .isEqualTo("€123.00")
+        assertThat(currencyFormatter.format(123412L, euro, Locale.US))
```

Comment:
Thoughts on passing in "EUR" here rather than a Currency instance? I think it would make the test flow a bit nicer and exercise a bit more code, since there are two `.format()` variants.

(And same for all of the other tests)

Reply:
So actually I am not sure that the results will actually be the same, I will check it out and report back some more information next week.

Reply:
I updated tests to include both